CI: ci-integration: TestUpdateUsedCIDIsReverted

## CI failure

Hit the following failure on a unrelated/freshly rebased PR:

```
--- FAIL: TestUpdateUsedCIDIsReverted (0.52s)
[...]

    logger.go:256: time=2025-09-08T14:31:18.041Z level=INFO source=/home/runner/work/cilium/cilium/operator/pkg/ciliumidentity/reconciler.go:191 msg="Creating CID" labels="map[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name:ns1 k8s:io.cilium.k8s.policy.cluster: k8s:io.kubernetes.pod.namespace:ns1 k8s:key-c:val-3]" ciliumIdentityName=9551
    controller_test.go:429: expected labels map[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name:ns1 k8s:io.cilium.k8s.policy.cluster: k8s:io.kubernetes.pod.namespace:ns1 k8s:key-b:val-2], got map[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name:ns1 k8s:io.cilium.k8s.policy.cluster: k8s:io.kubernetes.pod.namespace:ns1 k8s:key-c:val-3]

[...]
```

This reference this assert failed: https://github.com/cilium/cilium/blob/8e7f875a2f37a2d6fda64d33cdeeabd102856fa2/operator/pkg/ciliumidentity/controller_test.go#L429

Job link: https://github.com/cilium/cilium/actions/runs/17553955290/job/49853147234 / PR link: https://github.com/cilium/cilium/pull/41417

@cilium/operator

--- a/operator/pkg/ciliumidentity/controller_test.go
+++ b/operator/pkg/ciliumidentity/controller_test.go
@@ -419,17 +419,13 @@ func TestUpdateUsedCIDIsReverted(t *testing.T) {
 
 	require.EventuallyWithT(t, func(ct *assert.CollectT) {
 		cids := store.List()
-		assert.Len(ct, cids, 1)
-		if len(cids) != 1 {
+		if !assert.Len(ct, cids, 1) {
 			return
 		}
 		cid = cids[0]
 
-		if !cmp.Equal(cid.SecurityLabels, cid1.SecurityLabels) {
-			t.Fatalf("expected labels %v, got %v", cid.SecurityLabels, cid1.SecurityLabels)
-		}
+		assert.Equal(ct, cid1.SecurityLabels, cid.SecurityLabels)
 	}, WaitUntilTimeout, 100*time.Millisecond)
-
 }
 
 func TestDeleteUsedCIDIsRecreated(t *testing.T) {

